pygeo/parameterization/DVGeoCST.py

```diff
@@ -0,0 +1,1175 @@
+                N2=self.defaultDV[f"n2_{dvType}"],
+                dtype=self.dtype,
+            )
+
+            # Compute the quality of the fit by computing an L2 norm of the fit vs. the actual coordinates
+            xPts = self.foilCoords[self.idxFoil[dvType], self.xIdx]
+            yTE = self.thicknessTE / 2 if dvType == "upper" else -self.thicknessTE / 2
+            ptsFit = chord * self.computeCSTCoordinates(
+                xPts / chord,
+                self.defaultDV["n1_lower"],
+                self.defaultDV["n2_lower"],
+                self.defaultDV[dvType],
+                yTE,
+                dtype=self.dtype,
+            )
+            L2norm = np.sqrt(1 / ptsFit.size * np.sum((self.foilCoords[self.idxFoil[dvType], self.yIdx] - ptsFit) ** 2))
+
+            print(f"{dvType.capitalize()} surface")
+            print(f"    L2 norm of coordinates in dat file versus fit coordinates: {L2norm}")
+            print(f"    Fit CST coefficients: {self.defaultDV[dvType]}")
+
+    def addPointSet(self, points, ptName, boundTol=1e-10, **kwargs):
+        """
+        Add a set of coordinates to DVGeometry
+        The is the main way that geometry in the form of a coordinate list is given to DVGeometry
+        to be manipulated.
+
+        Parameters
+        ----------
+        points : array, size (N,3)
+            The coordinates to embed.
+        ptName : str
+            A user supplied name to associate with the set of coordinates.
+            This name will need to be provided when updating the coordinates or when
+            getting the derivatives of the coordinates.
+        boundTol : float, optional
+            Small absolute deviation by which the airfoil coordinates can exceed the initial
+            minimum and maximum x coordinates, by default 1e-10.
+        kwargs
+            Any other parameters are ignored.
+        """
+        # Convert points to the type specified at initialization (with isComplex) and store the points
+        points = points.astype(self.dtype)
+
+        # Check that all points are within the airfoil x bounds
+        if np.any(points[:, self.xIdx] < self.xMin - boundTol) or np.any(points[:, self.xIdx] > self.xMax + boundTol):
+            raise ValueError(
+                f'Points in the point set "{ptName}" have x coordinates outside'
+                + f"the min and max x values in the initial dat file ({self.xMin} and {self.xMax})"
+            )
+        self.updated[ptName] = False
+        self.points[ptName] = {
+            "points": points,
+            "xMax": self.xMax.copy(),
+            "xMin": self.xMin.copy(),
+            "thicknessTE": self.thicknessTE.copy(),
+        }
+
+        # Determine which points are on the upper and lower surfaces
+        self.points[ptName]["upper"], self.points[ptName]["lower"] = self._splitUpperLower(points)
+
+        # If debug mode is on, plot the upper and lower surface points
+        if self.debug:
```

Comment:
similar to the print, it might be better to do this while keeping in mind that the run can be parallel. maybe we can communicate everything back to root and do the plot once. it gets tricky because dvcon adds the same points on all procs whereas adflow will add a distributed pointset